### ISSUE TITLE
Cannot run with Junit 5

### ISSUE THREAD

**Issue by pat-lego** [tomakehurst/wiremock#1177]
## Bug reports

Wiremock continuously returns a `NotAuthorisedException` when running on JUnit Jupiter 5.5.1.

- [ ] Using version 2.24.1 of Wiremock
- [ ] Using the stack builder library within Junit tests to execute stubbed endpoints
- [ ] This test will return the same error every time

``` 
stubFor(get(urlEqualTo("/some/thing"))
                .willReturn(aResponse().withHeader("Content-Type", "text/plain").withBody("Hello world!")));
```
- [ ] Using maven 3.6.1
- [ ] Here is a summary of the error given back to me by Maven

```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.33 s <<< FAILURE! - in com.outreach.tests.helper.GetJWTTokenTest
[ERROR] invokeEndpoint  Time elapsed: 0.567 s  <<< ERROR!
com.github.tomakehurst.wiremock.security.NotAuthorisedException
        at com.outreach.tests.helper.GetJWTTokenTest.invokeEndpoint(GetJWTTokenTest.java:65)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors: 
[ERROR]   GetJWTTokenTest.invokeEndpoint:65 » NotAuthorised
[INFO]
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11.361 s
[INFO] Finished at: 2019-08-19T12:42:06-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project test-utilities: There are test failures.
[ERROR]
[ERROR] Please refer to C:\Users\Pat\Documents\repos\Outreach-Test-Helper\test-utilities\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

## Feature requests

Can we add support for Junit 5 in order to use the Wiremock library within Uni tests.

**Comment by tomakehurst** [comment]
Please can you post the code and config to reproduce this.

This is unlikely to be related to JUnit - it looks more like a port number problem to me.

**Comment by pat-lego** [comment]
Code: 

```
    @Test
    public void invokeEndpoint() throws Exception {
        String endpoint = "/bin/api/outreach/v2/login";

        stubFor(post(urlPathEqualTo(endpoint))
            .willReturn(aResponse()
            .withStatus(200)
            .withHeader("Content-Type", "application/json")
            .withBody("{\"jwt\": \"my_token\"}")));

        assertEquals("my_token", new TestableImpl().validInvocation());
    }
```

Config:
```
    @Rule
    public WireMockRule wireMockRule = new WireMockRule(options().port(8081));
```

**Comment by tomakehurst** [comment]
OK, I think the problem is that `@Rule` doesn't work with Jupiter.

You need to do one of the following:
* Use JUnit Vintage
* Use this: https://github.com/lanwen/wiremock-junit5
* Start and stop `WireMockServer` in your before and after methods

**Comment by pat-lego** [comment]
@tomakehurst sounds good I will test it out if it works I will submit a PR to the Read Me in order to inform others of this scenario.

Cheers.

**Comment by tomakehurst** [comment]
👍

**Comment by t1** [comment]
This issue can be closed → #684
